src/SelectMenu/hooks/KeyboardHook.js

```diff
@@ -0,0 +1,83 @@
+import {useEffect} from 'react'
+
+// adapted from details-menu web component https://github.com/github/details-menu-element
+function useKeyboardNav(details) {
+  const handleKeyDown = event => {
+    const closeDetails = () => {
+      details.current.removeAttribute('open')
```

Comment:
Is there a reason we're manipulating the DOM `open` attribute directly here instead of changing the value of the `open` state variable in the React component?

Reply:
The `open`state wasn't controlled until a few days ago when I changed some functionality, we could do that now

Reply:
updated!